website/public/js/controllers/rootCtrl.js

```diff
@@ -184,23 +190,62 @@ habitrpg.controller("RootCtrl", ['$scope', '$rootScope', '$location', 'User', '$
           $rootScope.charts[id] = (history.length == 0) ? false : !$rootScope.charts[id];
           if (task && task._editing) task._editing = false;
       }
+
       matrix = [[env.t('date'), env.t('score')]];
       _.each(history, function(obj) {
         matrix.push([moment(obj.date).format(User.user.preferences.dateFormat.toUpperCase().replace('YYYY','YY') ), obj.value]);
       });
       data = google.visualization.arrayToDataTable(matrix);
+
+      $rootScope.drawChart(id, data);
+    };
+
+    $rootScope.drawChart = function(id, data, options) {
```

Comment:
Does this need to be exposed onto `$rootScope`? Looks like it is only called internally.